malaria-ios/SocialShareable.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  SocialShareable.swift
+//  malaria-ios
+//
+//  Created by Eric Huang on 1/7/16.
+//  Copyright © 2016 Bruno Henriques. All rights reserved.
+//
+
+import Foundation
+import Social
+
+/// Protocol of objects that can be shared to social media.
+protocol SocialShareable {
+    var message: String { get }
+}
+
+// Example of Models Conforming to SocialShareable
+extension SocialShareable where Self: Stat {
+    
+    var message: String {
+        return ("My \(self.title) is \(self.attributeValue)")
+    }
+
+}
+
+extension MedicineLastTaken: SocialShareable {}
+extension Adherence: SocialShareable {}
+extension PillStreak: SocialShareable {}
+
+
```

Comment:
Just been reading a little bit of protocol extensions and figured it would be cool to use it here. Glad you like it :smile: